01_fib/NicholeL/fib.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func abs(args int) (res int) {
+	if args == 0 {
+		return 0
+	}
+	if args > 0 {
+		return args
+	}
+	return args * (-1)
+
+}
+
+func fib(args int) {
+	a, b := 0, 1
+	if abs(args) > 92 {
+		fmt.Fprint(out, "Fibonacci numbers greater than 92 not supported")
+		return
+	}
+	if abs(args) <= 1 {
+		fmt.Fprintln(out, args)
+	} else {
+		for index := 1; index <= abs(args); index++ {
+			if index < 2 {
+				fmt.Fprintln(out, b)
+			} else {
+				res := a + b
+				a, b = b, res
+				if args < 0 && index%2 == 0 {
+					fmt.Fprintln(out, res*(-1))
+				} else {
+					fmt.Fprintln(out, res)
+				}
+			}
+		}
+	}
```

Comment:
This function seems a bit more complicated than necessary. Here parts of our reference solution:

```go
func fib(n int) {
	sign := // 1 for positive n ; -1 for negative n; define inline
	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		// print n1
	}
}
```

Feel free to take, ignore or re-work.

Reply:
thanks for your suggestion